core/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -346,7 +353,8 @@ protected void processWay(ReaderWay way) {
             }
         }
 
-        IntsRef edgeFlags = encodingManager.handleWayTags(way, acceptWay, relationFlags);
+        List<CustomArea> customAreas = estimatedCenter == null ? emptyList() : customAreaIndex.query(estimatedCenter.lat, estimatedCenter.lon);
+        IntsRef edgeFlags = encodingManager.handleWayTags(way, acceptWay, relationFlags, customAreas);
```

Comment:
For now, we can also just put the custom areas into an artificial tag like `estimated_center`.